src/redis-pubsub.ts

```diff
@@ -11,10 +10,21 @@ export interface PubSubRedisOptions {
 
 export class RedisPubSub implements PubSubEngine {
 
-  constructor(options: PubSubRedisOptions = {}) {
+  constructor(options: PubSubRedisOptions = {}, publisher?: RedisClient, subscriber?: RedisClient) {
     this.triggerTransform = options.triggerTransform || (trigger => trigger as string);
-    this.redisPublisher = (IORedis as any).createClient(options.connection);
-    this.redisSubscriber = (IORedis as any).createClient(options.connection);
+    if (subscriber && publisher) {
+      this.redisPublisher = publisher;
+      this.redisSubscriber = subscriber;
+    } else {
+      try {
+        const IORedis = require('ioredis');
```

Comment:
Please add a test for that default behavior.

Reply:
Hey David, could you pleases elaborate on what you mean? I want to fix it.

Reply:
Oh sorry! missed this comment as it was on an outdated chunk of the commit.

From what I saw, on the current test suite, we don't have a test that checks for exactly this behavior. Meaning - don't provide a `options.subscriber` and `options.publisher` and check that the `RedisPubSub` is initialized properly.
You might have already created such test and I might have missed it so if so, please excuse me and point out the code bit that does that.

Thanks @amitshlo and I'll make sure to merge as soon as it's ready.